### DISCUSSION TITLE
Machine being interupted by unknown event `xstate.stop`

### DISCUSSION THREAD

**Discussion by lukewilde** [statelyai/xstate#3916] [Q&A] [answered]
Hello!

I'm in the process of upgrading `@xstate/react` from v1.6.3 -> v3.2.1. I've checked the docs for breaking changes ([2.0.0](https://github.com/statelyai/xstate/releases/tag/%40xstate%2Freact%402.0.0) and [3.0.0](https://github.com/statelyai/xstate/releases/tag/%40xstate%2Freact%403.0.0)) but couldn't see anything that seemed like it would need changes in our application.

However I am seeing some odd behaviour, one of our state machines makes an XHR request using `invoke`, however before the promise inside the `invoke.src` resolves, a state transition is occurring. To find out what event triggered it, I've added an exit action to console log the event and this is the culprit:

```
{type: 'xstate.stop'}
```
I'm having trouble finding out where that could be coming from. It looks internal but isn't reported by the xstate visualiser.

Any ideas?

Thanks in advance.

**Comment by Andarist** [comment]
Do you maybe use `autoForward` anywhere? Could you prepare a runnable repro case of the problem?

**Reply by lukewilde** [reply]
We're not using `autoForward`. I'll see if I can get a repro case put together. Thanks!

**Comment by lukewilde** [comment, marked as answer]
In our case, it turns out this was due to [`React.StrictMode`](https://react.dev/reference/react/StrictMode) causing our transition to get fired twice. This would have been fine, but we had a guard that was preventing entry back into the state after the first time it was accessed. Here's an example:

https://codesandbox.io/s/flamboyant-chatelet-y3emyv?file=/src/index.js
- I haven't included the guard
- Check the console logs for state info

Is the strict mode re-render reinitiating the instance returned by `useMachine`? I can't think of another way the `get` state would be entered twice.
